iceoryx_posh/include/iceoryx_posh/mepoo/chunk_header.hpp

```diff
@@ -57,57 +63,85 @@ struct alignas(32) ChunkHeader
     /// @return the pointer to the user-payload
-    void* userPayload() const noexcept;
+    void* userPayload() noexcept;
 
-    /// @brief Get the pointer to the user-header
-    /// @return the pointer to the user-header
-    template <typename T>
-    T* userHeader() const noexcept;
+    /// @brief Get a const pointer to the user-payload carried by the chunk
+    /// @return the const pointer to the user-payload
+    const void* userPayload() const noexcept;
 
     /// @brief Get a pointer to the `ChunkHeader` associated to the user-payload of the chunk
     /// @param[in] userPayload is the pointer to the user-payload of the chunk
     /// @return the pointer to the `ChunkHeader` or a `nullptr` if `userPayload` is a `nullptr`
-    static ChunkHeader* fromUserPayload(const void* const userPayload) noexcept;
+    static ChunkHeader* fromUserPayload(void* const userPayload) noexcept;
+
+    /// @brief Get a const pointer to the `ChunkHeader` associated to the user-payload of the chunk
+    /// @param[in] userPayload is the const pointer to the user-payload of the chunk
+    /// @return the const pointer to the `ChunkHeader` or a `nullptr` if `userPayload` is a `nullptr`
+    static const ChunkHeader* fromUserPayload(const void* const userPayload) noexcept;
 
     /// @brief Calculates the used size of the chunk with the ChunkHeader, user-heander and user-payload
     /// @return the used size of the chunk
     uint32_t usedSizeOfChunk() const noexcept;
 
+    /// @brief The size of the whole chunk, including the header
+    /// @return the chunk size
+    uint32_t chunkSize() const noexcept;
+
+    /// @brief The size of the chunk occupied by the user-payload
+    /// @return the user-payload size
+    uint32_t userPayloadSize() const noexcept;
+
+    /// @brief The unique identifier of the publisher the chunk was sent from
+    /// @return the id of the publisher the chunk was sent from
+    UniquePortId originId() const;
+
+    /// @brief A serial number for the sent chunks
+    /// @brief the serquence number of the chunk
+    uint64_t sequenceNumber() const;
+
   private:
+    template <typename T>
+    friend class popo::ChunkSender;
+
+    void setOriginId(UniquePortId originId);
+
+    void setSequenceNumber(uint64_t sequenceNumber);
+
     uint64_t overflowSafeUsedSizeOfChunk() const noexcept;
 
-    // END methods
+  private:
+    // the order of these members must be changed carefully and if this happens, the m_chunkHeaderVersion
+    // needs to be adapted in order to be able to detect incompatibilities between publisher/subscriber
+    // or record&replay, m_chunkSize and m_chunkHeaderVersion should therefore neither changed the type,
+    // nor the position
+
+    // size of the whole chunk, including the header
+    uint32_t m_chunkSize{0U};
+    uint8_t m_chunkHeaderVersion{CHUNK_HEADER_VERSION};
+    // reserved for future functionality and used to indicate the padding bytes; currently not used and set to `0`
```

Comment:
Why not use an array for the reserved area? The size could be a defined constant. We might need to adapt it later but this will break ABI compatibility anyway.

Reply:
done